includes/class-wc-payments.php

```diff
@@ -510,6 +512,25 @@ public static function register_checkout_gateway( $payment_method_registry ) {
 		$payment_method_registry->register( new WC_Payments_Blocks_Payment_Method() );
 	}
 
+	/**
+	 * Handles upgrade routines.
+	 */
+	public static function install() {
+		self::add_woo_admin_notes();
+		if ( WCPAY_VERSION_NUMBER !== get_option( 'woocommerce_woocommerce_payments_version' ) ) {
```

Comment:
I think it would be better to use [`version_compare`](https://www.php.net/manual/en/function.version-compare.php) here to check for a newer version?

Or maybe this should also happen when the plugin is downgraded? The action name, `woocommerce_woocommerce_payments_updated` confuses the logic a bit here. Maybe a better name would be `woocommerce_woocommerce_payments_version_changed` or something along those lines?

Reply:
The version check and the action name is another instance of a reused code from a previous implementation in Stripe 🙂

I switched this to `version_compare`.

> Or maybe this should also happen when the plugin is downgraded? 

This shouldn't happen when the plugin is downgraded.

> The action name, woocommerce_woocommerce_payments_updated confuses the logic a bit here.

The idea is to call this action only when the plugin is updated. Switching to `version_changed` might cause confusion, I guess. What do you think?

Reply:
> The idea is to call this action only when the plugin is updated. Switching to `version_changed` might cause confusion, I guess. What do you think?

Yeah, I should've been clearer. What I meant to say was that `updated` is perfectly fine if the action is called when the plugin has been _updated_, whereas `version_changed` would be better if the action is called whenever the plugin version _changes_.

Since the action is only supposed to be called when the plugin is updated, the current `updated` name is perfectly fine as it is ☺️